rtems/Mutex.java

```diff
@@ -22,16 +26,20 @@ public void lock() throws InterruptedException{
 		try{
 			if((holder!=null) && (holder!=thisThread))
 			{
-				if(priorityRaiseFilter(thisThread.getPriority()))
+				if(priorityRaiseFilter(thisThread.currentPriority))
 				{
 					//1. Update priority of holder thread
-					updatePriority();
+					updatePriority(thisThread.currentPriority);
+					//for solution to nested mutex problem call below
+					//updateRecPriority(thisThread.currentPriority);
 					//2. Re-enqueue holder thread with modified priority if its waiting 
-					renqueue();
+					reEnqueue();
 				}
 				thisThread.state = Thread.State.WAITING;
 				while(!(thisThread.state==Thread.State.RUNNABLE)){
-					cv1.await();
```

Comment:
I know this isn't a change in this PR but:
`while (thisThread.state != Thread.State.RUNNABLE`